### PR TITLE
pulls approved changes into 4-GET-/api/articles/parametric

### PR DESCRIPTION
**Pull request by swells2020**

### CODE REVIEW

File: controllers/controller.js [hunk 1]
```diff
@@ -12,6 +12,15 @@ exports.getApi = (request, response) => {
         })
 };
 
+exports.getTopics = (request, response) => {
+    fetchTopics().then(({ rows }) => {
+        response.send({ rows });
+    })
```
Comment (Poonam-raj) [inline controllers/controller.js:18]: Should be sending back on the key of `topics` not `rows`

File: __tests__/server.test.js [hunk 2]
```diff
@@ -39,44 +39,64 @@ describe('app', () => {
...
-                .get('/api/articles/thirteen')
-                .expect(400)
-                .then(({ body: { message } }) => {
-                    expect(message).toBe('400: bad request');
-                })
-        })
+        return request(app)
+            .get('/api/articles/1')
+            .expect(200)
+            .then(({ body: { rows } }) => {
```
Comment (Poonam-raj) [inline __tests__/server.test.js:77]: should be sending data back on the key of `article` not `rows`

Comment (Poonam-raj) [reply __tests__/server.test.js:77]: Also the returned article should not be nested in an array aka should come back as `body: {articles: {...}}`

File: __tests__/server.test.js [hunk 3]
```diff
@@ -39,44 +39,64 @@ describe('app', () => {
...
+                expect(rows[0].body).toEqual(testObject1.body)
+                expect(rows[0].votes).toEqual(testObject1.votes)
+            })
+    })
+    test('tests the error handling for bad parametric paths', () => {
+        return request(app)
+            .get('/api/articles/13')
+            .expect(404)
+            .then(({ body: { message } }) => {
+                expect(message).toBe('404: invalid path');
```
Comment (Poonam-raj) [inline __tests__/server.test.js:91]: invalid path not the most appropriate message here - more likely we want to see `404: article not found`

**Review by Poonam-raj — COMMENTED** [review]
404 message, the data structure being sent back on for GET article/articleID and the key you send information back for article and topics 
Please amend these on a new branch and make a new PR with those changes :D

File: controllers/controller.js (hunk 1 above)
Comment (swells2020) [reply controllers/controller.js:18]: Updated, thanks.

File: __tests__/server.test.js (hunk 2 above)
Comment (swells2020) [reply __tests__/server.test.js:77]: Updated, thanks.

**Pull request merged**